internal/workflow/processing.go

```diff
@@ -908,23 +908,30 @@ func (w *ProcessingWorkflow) transferAM(ctx temporalsdk_workflow.Context, tinfo
 		return err
 	}
 
-	// Zip PIP.
-	activityOpts := withActivityOptsForLocalAction(ctx)
-	var zipResult archivezip.Result
-	err = temporalsdk_workflow.ExecuteActivity(
-		activityOpts,
-		archivezip.Name,
-		&archivezip.Params{SourceDir: tinfo.TempPath},
-	).Get(activityOpts, &zipResult)
-	if err != nil {
-		return err
-	}
+	// Zip PIP, if necessary.
+	var sourcePath string
+	if w.cfg.AM.TransferType == "zipped bag" {
+		// Zip PIP.
+		activityOpts := withActivityOptsForLocalAction(ctx)
+		var zipResult archivezip.Result
+		err = temporalsdk_workflow.ExecuteActivity(
+			activityOpts,
+			archivezip.Name,
+			&archivezip.Params{SourceDir: tinfo.TempPath},
+		).Get(activityOpts, &zipResult)
+		if err != nil {
+			return err
+		}
 
-	tinfo.SendToFailed.Path = zipResult.Path
-	tinfo.SendToFailed.ActivityName = activities.SendToFailedPIPsName
+		tinfo.SendToFailed.Path = zipResult.Path
+		tinfo.SendToFailed.ActivityName = activities.SendToFailedPIPsName
+		sourcePath = zipResult.Path
```

Comment:
Huh, this is totally not because of your changes but I think we should be registering the zipfile path for cleanup here with `cleanup.registerPath()`. :open_mouth: You'll need to pass the cleanup struct into `transferAM()` like we are doing for `transferA3M()`.

Reply:
I think you can just reassign `tinfo.TempPath = zipResult.Path` here instead of using `sourcePath`.